include/nfl/core.hpp

```diff
@@ -79,69 +79,54 @@ poly<T, Degree, NbModuli>::poly(It first, It last, bool reduce_coeffs) {
 
 template<class T, size_t Degree, size_t NbModuli>
 void poly<T, Degree, NbModuli>::set(value_type v, bool reduce_coeffs) {
-  set({v}, reduce_coeffs);
+  if (v == 0) {
+    // CRITICAL: the object must be 32-bytes aligned to avoid vectorization issues
+    assert((unsigned long)(this->_data) % 32 == 0);
+    for (auto* iter = begin(); iter<end(); ++iter) *iter = 0;
+  }
```

Comment:
This modification makes the setter way faster in applications (because by default, reduce_coeff = true).